core-rust/state-manager/src/store/traits.rs

```diff
@@ -174,14 +174,19 @@ pub mod transactions {
         LocalTransactionReceipt,
     };
 
-    #[enum_dispatch]
-    pub trait QueryableTransactionStore {
-        fn get_committed_transaction_bundles(
+    pub trait IterableTransactionStore {
+        type CommittedTransactionBundleIterator<'a>: Iterator<Item = CommittedTransactionBundle>
+        where
+            Self: 'a;
+
+        fn get_committed_transaction_bundle_iter(
             &self,
-            start_state_version_inclusive: u64,
-            limit: usize,
-        ) -> Vec<CommittedTransactionBundle>;
+            from_state_version: u64,
+        ) -> Self::CommittedTransactionBundleIterator<'_>;
```

Comment:
If we put the trait as `Box<dyn Iterator<Item = CommittedTransactionBundle> + '_>` here and dispense with the associated type/GAT then in a few cases (eg db.rs and with the account index trait) we could just return an iterator direct from eg RocksDB rather than create our own iterator struct. EG see `list_entries` in the `SubstateDatabase` trait.

The slightly lower performance would hardly matter in this case compared to the other 10000x larger overheads from the transaction mapping.

Reply:
Sure. Let's go with `dyn`.